pkg/framework/test/doc.go

```diff
@@ -0,0 +1,52 @@
+/*
+
+Package test implements a integration testing framework for kubernetes.
+
+It provides a kubernetes API you can connect to and test your
+kubernetes client implementations against. The lifecycle of the components
```

Comment:
I'd really like if we could not phrase this framework as a "client test framework". One could want to test a new etcd version by switching the binary. One could decide to test the apiserver, an extension apiserver, an aggregator, a FooBar component somewhere, etc. I think these are built as reusable pieces that can be composed to test any component in a kubernetes cluster.